cluster/src/test/scala/io/snappydata/cluster/PreparedQueryRoutingSingleNodeSuite.scala

```diff
@@ -470,6 +470,329 @@ class PreparedQueryRoutingSingleNodeSuite extends SnappyFunSuite with BeforeAndA
+      // scalastyle:on println
+      assert(index == 3)
+      assert(cacheMap.size() == 1)
+      close(prepStatement0)
+
+      val prepStatement1 = conn.prepareStatement(s"select a + ?, d from double_tab")
+      assert(cacheMap.size() == 1)
+      prepStatement1.setInt(1, 2)
+      update = prepStatement1.executeQuery()
+      index = 0
+      while (update.next()) {
+        val i = update.getInt(1)
+        val j = update.getBigDecimal(2)
+        // scalastyle:off println
+        println(s"3-row($index) $i $j")
+        // scalastyle:on println
+        index += 1
+        assert(i > 2 && i < 6)
+      }
+      // scalastyle:off println
+      println(s"3-Number of rows read " + index)
+      // scalastyle:on println
+      assert(index == 3)
+      assert(cacheMap.size() == 2)
+
+      prepStatement1.setInt(1, 3)
+      update = prepStatement1.executeQuery()
+      index = 0
+      while (update.next()) {
+        val i = update.getInt(1)
+        val j = update.getBigDecimal(2)
+        // scalastyle:off println
+        println(s"4-row($index) $i $j")
+        // scalastyle:on println
+        index += 1
+        assert(i > 3 && i < 7)
+      }
+      // scalastyle:off println
+      println(s"4-Number of rows read " + index)
+      // scalastyle:on println
+      assert(index == 3)
+      assert(cacheMap.size() == 2)
+      close(prepStatement1)
+
+      val prepStatement2 = conn.prepareStatement(s"select a," +
+          s" d from double_tab where UPPER(s) = ?")
+      assert(cacheMap.size() == 2)
+      prepStatement2.setString(1, "1A")
+      update = prepStatement2.executeQuery()
+      index = 0
+      while (update.next()) {
+        val i = update.getInt(1)
+        val j = update.getString(2)
+        // scalastyle:off println
+        println(s"5-row($index) $i $j")
+        // scalastyle:on println
+        index += 1
+        assert(i == 1)
+      }
+      // scalastyle:off println
+      println(s"5-Number of rows read " + index)
+      // scalastyle:on println
+      assert(index == 1)
+      assert(cacheMap.size() == 3)
+
+      prepStatement2.setString(1, "2B")
+      update = prepStatement2.executeQuery()
+      index = 0
+      while (update.next()) {
+        val i = update.getInt(1)
+        val j = update.getString(2)
+        // scalastyle:off println
+        println(s"6-row($index) $i $j")
+        // scalastyle:on println
+        index += 1
+        assert(i == 2)
+      }
+      // scalastyle:off println
+      println(s"6-Number of rows read " + index)
+      // scalastyle:on println
+      assert(index == 1)
+      assert(cacheMap.size() == 3)
+      close(prepStatement2)
+
+      var prepStatement3: PreparedStatement = null
+      try {
+        prepStatement3 = conn.prepareStatement(s"select * from double_tab" +
+            s" where round(d, ?) < ?")
+        fail()
+      } catch {
+        case e: UnsupportedOperationException =>
+        case e: SQLException =>
```

Comment:
Check if the message you send is there in the exception message or not. i.e. '?' mark should not be there in the function argument.